python/ray/serve/tests/test_target_capacity.py

```diff
@@ -373,6 +380,358 @@ def test_autoscaling_scale_to_zero(
+            target_num_ongoing_requests_per_replica=1,
+            metrics_interval_s=0.01,
+            look_back_period_s=0.01,
+            upscale_delay_s=0.01,
+            downscale_delay_s=0.01,
+        ),
+        graceful_shutdown_timeout_s=0,
+    ).bind()
+
+
+class TestTargetCapacityUpdateAndServeStatus:
+    def check_num_running_replicas(
+        self, expected_num_running_replicas: int, app_name: str, deployment_name: str
+    ) -> bool:
+        deployment = serve.status().applications[app_name].deployments[deployment_name]
+        num_running_replicas = deployment.replica_states.get(ReplicaState.RUNNING, 0)
+        assert num_running_replicas == expected_num_running_replicas
+        return True
+
+    def apply_config_and_check_status(
+        self,
+        client: ServeControllerClient,
+        target_capacity: Optional[float],
+        config: ServeDeploySchema,
+        app_name: str,
+        deployment_name: str,
+        check_statuses: bool = True,
+    ):
+        """Applies config with specified target_capacity."""
+
+        config.target_capacity = target_capacity
+        client.deploy_apps(config)
+        wait_for_condition(lambda: serve.status().target_capacity == target_capacity)
+        if check_statuses:
+            wait_for_condition(
+                lambda: serve.status().applications[app_name].status
+                == ApplicationStatus.DEPLOYING
+            )
+            wait_for_condition(
+                lambda: serve.status()
+                .applications[app_name]
+                .deployments[deployment_name]
+                .status
+                in [
+                    DeploymentStatus.UPDATING,
+                    DeploymentStatus.UPSCALING,
+                    DeploymentStatus.DOWNSCALING,
+                ]
+            )
+            assert serve.status().applications[app_name].deployments[
+                deployment_name
+            ].status_trigger in [
+                DeploymentStatusTrigger.CONFIG_UPDATE_STARTED,
+                DeploymentStatusTrigger.AUTOSCALING,
+            ]
+
+    def start_replicas_and_check_status(self, signal, app_name: str):
+        ray.get(signal.send.remote())
+        wait_for_condition(
+            lambda: serve.status().applications[app_name].status
+            == ApplicationStatus.RUNNING
+        )
+        ray.get(signal.send.remote(clear=True))
+
+    def test_static_num_replicas_target_capacity_update(
+        self, shutdown_ray_and_serve, client: ServeControllerClient
+    ):
+        """Check how Serve's status updates when target_capacity changes."""
+
+        app_name = "controlled_app"
+        deployment_name = "controlled"
+        num_replicas = 20
+
+        signal = SignalActor.options(
+            name="lifecycle_signal", namespace=SERVE_NAMESPACE
+        ).remote()
+        _ = SignalActor.options(
+            name="request_signal", namespace=SERVE_NAMESPACE
+        ).remote()
```

Comment:
why do we need to start this if we don't use it?

Reply:
if it's because the deployment fails without it existing, you can have that just catch the `ValueError`

Reply:
Good idea, I caught the `ValueError` as suggested.